rust/processor/src/processors/econia_processor.rs

```diff
@@ -618,14 +651,76 @@ impl ProcessorTrait for EconiaTransactionProcessor {
+            for change in &info.changes {
+                match change.change.as_ref().expect("No transaction changes") {
+                    Change::WriteResource(resource) => {
+                        if resource.r#type.as_ref().expect("No resource type")
+                            == &market_accounts_type
+                        {
+                            let data: serde_json::Value = serde_json::from_str(&resource.data)
+                                .expect("Failed to parse MarketAccounts");
+                            market_account_handles.push(MarketAccountHandle {
+                                user: resource.address.clone(),
+                                handle: data["map"]["handle"].as_str().unwrap().to_string(),
+                            })
+                        }
+                    },
+                    Change::WriteTableItem(write) => {
+                        let table_data = write.data.as_ref().expect("No WriteTableItem data");
+                        if table_data.value_type
+                            != format!("{}::user::MarketAccount", econia_address)
+                        {
+                            continue;
+                        }
+                        let table_key: serde_json::Value = serde_json::from_str(&table_data.key)
+                            .expect("Failed to parse market account ID");
+                        let market_account_id =
+                            u128::from_str(table_key.as_str().unwrap()).unwrap();
+                        let data: serde_json::Value = serde_json::from_str(&table_data.value)
+                            .expect("Failed to parse MarketAccount");
+                        balance_updates.push(BalanceUpdate {
+                            txn_version: txn_version.into(),
+                            handle: write.handle.to_string(),
+                            market_id: ((market_account_id >> SHIFT_MARKET_ID) as u64).into(),
+                            custodian_id: ((market_account_id & HI_64) as u64).into(),
+                            time,
+                            base_total: u64::from_str(data["base_total"].as_str().unwrap())
+                                .unwrap()
+                                .into(),
+                            base_available: u64::from_str(data["base_available"].as_str().unwrap())
+                                .unwrap()
+                                .into(),
+                            base_ceiling: u64::from_str(data["base_ceiling"].as_str().unwrap())
+                                .unwrap()
+                                .into(),
+                            quote_total: u64::from_str(data["quote_total"].as_str().unwrap())
+                                .unwrap()
+                                .into(),
+                            quote_available: u64::from_str(
+                                data["quote_available"].as_str().unwrap(),
+                            )
```

Comment:
I'd like to avoid the use of `unwrap` just like above in the code. I'd add a function just like `opt_value_to_{string,bool,big_decimal}` but for a u64 from str just for this (`opt_value_str_to_u64`) that just returns a result.

Reply:
Unwraps removed per 306a3cb